torch/package/package_importer.py

```diff
@@ -170,14 +173,21 @@ def load_pickle(self, package: str, resource: str, map_location=None) -> Any:
         restore_location = _get_restore_location(map_location)
         loaded_storages = {}
         loaded_reduces = {}
+        storage_context = torch._C.StorageContextTracker()
 
         def load_tensor(data_type, size, key, location, restore_location):
-            name = f".data/{key}.storage"
+            name = f"{key}.storage"
             dtype = data_type(0).dtype
 
-            storage = self.zip_reader.get_storage_from_record(
-                name, size, dtype
-            ).storage()
+            if storage_context.has_tensor_impl(name):
+                storage = storage_context.get_tensor_impl(name).storage()
+            else:
+                tensor = self.zip_reader.get_storage_from_record(
+                    ".data/" + name, size, dtype
+                )
+                if isinstance(self.zip_reader, torch._C.PyTorchFileReader):
```

Comment:
The `MockZipReader`'s returned 'storages' are of type 'torch.package._mock_zipreader._HasStorage' instead of an actual tensor , which is incompatible with the `StorageContextTracker''s requirement of needing a tensor passed between cpp and python. Due to the issue of TS also not being compatible with the `MockZipReader`, I don't think this will be a real issue

Reply:
maybe add a comment describing what this is actually checking (for the case where `zip_reader` is a `MockZipReader`.